app/main/views/suppliers.py

```diff
@@ -119,6 +119,38 @@ def view_signed_agreement(supplier_id, framework_slug):
     )
 
 
+@main.route('/suppliers/agreements/<agreement_id>/on-hold', methods=['POST'])
+@login_required
+@role_required('admin-ccs-sourcing')
+def put_signed_agreement_on_hold(agreement_id):
+    data_api_client.put_signed_agreement_on_hold(agreement_id, current_user.email_address)
+
+    organisation = request.args['organisation']
+    flash('The agreement for {} was put on hold.'.format(organisation), 'message')
+
+    return redirect(url_for(
+        '.list_agreements',
+        framework_slug='g-cloud-8')
+    )
+
+
+@main.route('/suppliers/agreements/<agreement_id>/approve', methods=['POST'])
+@login_required
+@role_required('admin-ccs-sourcing')
+def approve_agreement_for_countersignature(agreement_id):
+    data_api_client.approve_agreement_for_countersignature(agreement_id,
+                                                           current_user.email_address,
+                                                           current_user.id)
+
+    organisation = request.args['organisation']
+    flash('A confirmation was sent to {}.'.format(organisation), 'message')
```

Comment:
I don't think this message is right - no-one is sent anything at this point.  How about `'The agreement for {} was approved for countersigning.`?  Or maybe @laurenceberry has better words?